support/logging/file.go

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright (c) 2018 Cavium
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package logging
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/edgexfoundry/edgex-go/support/domain"
+)
+
+const (
+	rmFileSuffix string = ".tmp"
+)
+
+type fileLog struct {
+	filename string
+	out      io.WriteCloser
+}
+
+func (fl *fileLog) add(le support_domain.LogEntry) {
+	if fl.out == nil {
+		var err error
+		fl.out, err = os.OpenFile(fl.filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			//fmt.Println("Error opening log file: ", fl.filename, err)
+			fl.out = nil
+			return
+		}
+	}
+
+	res, err := json.Marshal(le)
+	if err != nil {
+		return
+	}
+	fl.out.Write(res)
+	fl.out.Write([]byte("\n"))
```

Comment:
Why do we null-terminate here? If it needs to be added to the string, maybe it is better to do it before marshaling.

Reply:
Several reasons:
- This way the file log could be read by humans, having a line between each json make the file 'somewhat' readable
- The \n is used by bufio scanner as delimiter to parse each entry in the logs to filter it later. I'm sure there is a way to read all the json without any delimiter between them, but it seemed like a good enough solution to start with.